emcc.py

```diff
@@ -1242,6 +1242,12 @@ def check(input_file):
       if shared.Settings.WASM:
         shared.Settings.BINARYEN = 1 # these are synonyms
 
+        # When only targeting wasm, the .asm.js file is not executable, so is treated as an intermediate build file that can be cleaned up.
+        if shared.Building.is_wasm_only():
+          asm_target = asm_target.replace('.asm.js', '.temp.asm.js')
```

Comment:
Ok, about making two max-perf builds that can be picked at runtime, we should probably do something like provide a little JS people can put in their HTML, that loads the right one. Then people just need to make two builds and add that JS. Another option is to add an emcc flag "FALLBACK_ASMJS_BUILD=x.js" and then we would make html that loads that asm.js build. Perhaps emcc could also invoke emcc to build that asm.js build, but not sure how much it's worth us doing.

Yeah, fair enough about altering existing files. Honestly I don't worry about it too much ;) but sure, lgtm.